src/test/scala/org/kframework/minikore/NodeTests.scala

```diff
@@ -0,0 +1,205 @@
+package org.kframework.minikore
+
+import org.junit.Test
+import org.kframework.minikore.interfaces.pattern._
+import org.kframework.minikore.interfaces.tree._
+import org.kframework.minikore.implementation.DefaultBuilders
+import org.kframework.minikore.interfaces.build
+
+
+class NodeTest {
+
+  object TestFunctions {
+
+    def size(p: Pattern): Int = p match {
+      case Node(c: Seq[Pattern]) => c.map(size).sum
+      case Leaf(_) => 1
+    }
+
+    def getNode0Count(p: Pattern): Int = p match {
+      case Node0() => 1
+      case Node(c: Seq[Pattern]) => c.map(getNode0Count) sum
```

Comment:
@grosu, there is nothing wrong with `c.map(f)` as opposed `c map f`. I think the idea for the dotless syntax is to make it clear/easier to read. Other languages, including Haskell, have taken the same approach. 

Regarding its particular meaning, Scala is not only functional but also object-oriented, so I'm not sure the conventions from other languages can apply directly. The way I see it, in Haskell, function application is the most common operation, so it chose to use space for this operation. In Scala, method call is the most common operation so it chose to use space for it. 

Still, if seeing an object as a function taking as parameters the method name and something else, one could see Scala's syntax fit into the functional paradigm as well. `c map` means apply function `c` to argument `map`, then apply to result to argument `f`.